### DISCUSSION TITLE
Form (AbstractForm) inherit at other page content

### DISCUSSION THREAD

**Discussion by kilimetr** [wagtail/wagtail#8968] [Ideas]
Hello everybody,

please, is there any possible way to have a Form inherit on certain page which already contains a content? My idea is that I have separate Form app which I want to load on certain pages (chosen by me). 

I need to send a customer data on email - so I am using form for sending email with hiddeninput where i manually load the content. I want to only be visible button send (submit) to customer. If there is any other better and simple way, please, share you knowledge and ideas.

thank you.

**Comment by lb-** [comment]
Does this Stackoverflow question help? https://stackoverflow.com/questions/42673180/how-can-i-add-a-form-made-by-formbuilder-to-every-page-in-wagtail

I would recommend you have a stand-alone page which contains the 'form' and then just add some kind of toggle to your other page models that enables this. Then embed the form (as per the question above) when this toggle is enabled.

The POST would actually go to the form page but you can then redirect the user page to the page they were at if needed.

This docs guide has some helpful info about how to use form page with routablePageMixin to redirect the user after submit https://docs.wagtail.org/en/latest/advanced_topics/formbuilder_routablepage_redirect.html

**Reply by kilimetr** [reply]
Hey Ben,

thank you for your help.

I like more the way to include RoutableMixin, which would grad the data from basepage and fed the formapp. Do you know how to execute the formapp to load the data in (the action after which happends after submit button is clicked by user)?

**Reply by lb-** [reply]
Hey @kilimetr I don't really understand this new question sorry.

However the next release of Wagtail may provide a nice solution here. 

In the mean time I recommend you read the docs section on how to do some more complex integrations of the Form builder to see if it helps for your question above. 

https://docs.wagtail.org/en/stable/reference/contrib/forms/customisation.html

**Comment by lb-** [comment]
The next release of Wagtail will introduce a solution for this. 

https://github.com/wagtail/wagtail/pull/9093

Which is a rebase of the original PR
https://github.com/wagtail/wagtail/pull/2900

**Comment by kilimetr** [comment]
Hey Ben,

I need to trigger an action after the button is hit. I need to send some info to customer's email. Therefore I was think to modify a forms which are sending the email in order to avoid any other programming (save some time). But this way is not that easy as i thought.

However, is it possible to program this button at wagtail? or it will be better to program it at django (as app) and connected to wagtail project?

**Reply by lb-** [reply]
Just checking. Have you read this example in the docs?

https://docs.wagtail.org/en/stable/reference/contrib/forms/customisation.html#custom-send-mail-method

It explains exactly how to do extra 'stuff' when en email is triggered from an Email Form Page.

Does that help?

**Reply by lb-** [reply]
It's also good to clarify what you mean by 'when the button is clicked'. There is the clientside click (JavaScript) and the subsequent form submission which is a POST request to the page. 

That POST request goes through the form logic, adds the submission row and then does the email sending (if using the right abstract page). 

It's usually simpler to work with the web requests as built into Django and then into Wagtail. Unless you have a reason to do something else.
